etca_asm/core.py

```diff
@@ -98,64 +105,98 @@ def core_init(context):
                                   if (e := potential_extensions[a]).default_on]
     context.modes = set()
     context.ip = 0x8000
-    context.labels = {}
-    context.last_labels = ['']
-    context.missing_labels = set()
-    context.changed_labels = set()
-    context.resolve_label = partial(_resolve_label, context)
+    context.symbols = {}
+    context.symbol_path = ['']
+    context.missing_symbols = set()
+    context.changed_symbols = set()
+    context.illegal_symbols = set()
+    context.resolve_symbol = partial(_resolve_symbol, context)
     for e in potential_extensions.values():
         if e.init is not None and e is not core:
             e.init(context)
     context.reload_extensions()
 
 
+def oneof(*names):
+    names = sorted(names, key=len, reverse=True)
+    return f"({'|'.join(names)})"
+
+
+_WORD_SIZES = {
+    '.half': 1,
+    '.word': 2,
+    '.dword': 4,
```

Comment:
Should we add this just so we don't forget?

Reply:
And maybe something for arrays/strings?

Reply:
Let's do that separately, this PR is already fairly scatterbrained.